src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -325,6 +325,10 @@ class FastSync(
         .addPendingNodes(hashesToRequest.flatten)
         .copy(downloadedNodesCount = syncState.downloadedNodesCount + nodeData.values.size)
 
+
+      appStateStorage.putPulledStateNodes(syncState.downloadedNodesCount)
```

Comment:
I wonder, do we need to save those to leveldb? We already do that as a part of sync state (persistSyncState). Maybe we can load it from there on startup and keep as a var somewhere? Or maybe EthService could just ask FastSync for `SyncState`?

Reply:
good call, it is just case of using existing `FastSyncStorage`  to get currently persisted state.

Reply:
this should work too although keep in mind this state is only persisted every 1 minute or so (I think)

Reply:
yep every minute, but I think we will be good with it, we just want to give the user some feedback that something is happening